### PR TITLE
components, src: avoid cpu quota limitation contamination 

### LINKED ISSUE #13084: Avoid mutual contamination of cpu limitation of foreground and background cpu quota
### What version of TiKV are you using?

master
### What operating system and CPU are you using?

any
### Steps to reproduce

The repro is quite simple:
Enable cpu quota limit for foreground requests, background requests like auto analyze will spend the overhead to collection cpu time usage and vice verse. Which is improper and brings overhead.
### What did you expect?
No such contamination.
### What did happened?
Contamination occurred.

### PR DESCRIPTION
**Pull request by BornChanger**
Signed-off-by: BornChanger <dawn_catcher@126.com>

### What is changed and how it works?

Issue Number: Close #13084 

What's Changed:

```
Sample build didn't check if cpu quota is from foreground or background. 
So, if either cpu quota limit is set, cpu sampling would be conducted for both 
foreground and background.  It's not expected and brought overhead.  We 
need to distinguish the cpu quota limit sources.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)
Debug to make sure no such contamination any more.

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by BornChanger** [comment]
@tabokie @Connor1996 please have a review to this quick fix.

**Review by tabokie — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

File: src/coprocessor/statistics/analyze.rs [hunk 1]
```diff
@@ -359,7 +359,7 @@ impl<S: Snapshot> RowSampleBuilder<S> {
             columns_info,
             column_groups: req.take_column_groups().into(),
             quota_limiter,
-            is_quota_auto_tune,
+            is_auto_analyze: is_quota_auto_tune,
```
Comment (sticnarf) [inline src/coprocessor/statistics/analyze.rs:362]: Could you please explain a bit more about the name? To my understanding, auto analyze is the source type of TiDB analyze. How is it related to the quota auto tune?

I can understand the old name, but the new name is a bit confusing to me.

Comment (BornChanger) [reply src/coprocessor/statistics/analyze.rs]: fixed the typo.

**Review by sticnarf — APPROVED** [review]
LGTM

**Pull request merged**